webapp/src/Security/UserStateUpdater.php

```diff
@@ -31,7 +31,15 @@ public static function getSubscribedEvents(): array
     public function updateUserState(AuthenticationSuccessEvent $event): void
     {
         if ($event->getAuthenticationToken() && ($user = $event->getAuthenticationToken()->getUser()) && $user instanceof User) {
-            $user->setLastLogin(Utils::now());
+            $firewallName = 'main';
+            if (method_exists($event->getAuthenticationToken(), 'getFirewallName')) {
+                $firewallName = $event->getAuthenticationToken()->getFirewallName();
+            }
+            if ($firewallName === 'main') {
+                $user->setLastLogin(Utils::now());
+            } elseif ($firewallName === 'api') {
+                $user->setLastApiLogin(Utils::now());
+            }
```

Comment:
Consider adding an `else` so we can detect if we add new firewalls later?

Reply:
We *have* more firewalls, like `metrics`. What should we do with them? At first I had main => lastlogin else => lastapilogin, but does that make sense?

Reply:
metrics is also over the API so for that specific case it would work, but with another firewall this would not hold anymore. Maybe this is indeed the best solution.

Reply:
Metrics has a different firewall though. Which I think we should add here, let me do it